### PR TITLE
587 store table obj correctly

### PR DESCRIPTION
**Pull request by UlrichKu**
| Q             | A
| ------------- | ---
| Branch        | 6.3.x
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed issues  | chameleon-system/chameleon-system#587
| License       | MIT

### CODE REVIEW

File: src/ShopBundle/objects/TCMSListManager/TCMSListManagerShopArticles.php [hunk 1]
```diff
@@ -25,21 +25,18 @@ protected function PostCreateTableObjectHook()
...
-
-        $filterArticleType = $this->tableObj->getCustomSearchFieldParameter('filterArticleType');
+        $request = $this->getRequest();
 
-        if (null === $filterArticleType) {
-            $filterArticleType = 'all';
+        if ($request->query->has('is_active')) {
+            $isActive = $request->query->get('is_active');
+        } else {
+            $isActive = $this->tableObj->_postData['is_active'] ?? '';
```
Comment (rnuernberg) [inline src/ShopBundle/objects/TCMSListManager/TCMSListManagerShopArticles.php:33]: The value used to be accessed via `getCustomSearchFieldParameter` - why the change to _postData?

File: src/ShopBundle/objects/TCMSListManager/TCMSListManagerShopArticles.php [hunk 2]
```diff
@@ -25,21 +25,18 @@ protected function PostCreateTableObjectHook()
...
         }
 
-        $request = $this->getRequest();
         if (true === $request->query->has('filterArticleType')) {
             $filterArticleType = $request->query->get('filterArticleType');
+        } else {
+            $filterArticleType = $this->tableObj->_postData['filterArticleType'] ?? 'all';
```
Comment (rnuernberg) [inline src/ShopBundle/objects/TCMSListManager/TCMSListManagerShopArticles.php:39]: some question here

File: src/ShopBundle/objects/TCMSListManager/TCMSListManagerShopArticles.php (hunk 1 above)
Comment (UlrichKu) [reply src/ShopBundle/objects/TCMSListManager/TCMSListManagerShopArticles.php:33]: The custom search field parameters seem to be an (old?) unused mechanism that is only found here.

Other similar locations (e. g. for "searchTerm") are solely using "_postData" (which actually translates to "all data for the current page including stored ones".

With the custom search field parameters used this is a 4-times indirection on the "page data".
With "_postData" it is "only" 2-times indirection.
